### PR TITLE
Include Registration Option for SUSE distributions

### PR DESCRIPTION
**Pull request by diegoakechi**
Include support to register the system to the SUSE Customer Center
Include also support to register additional modules and extensions

### CODE REVIEW

File: salt/default/registration.sls [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% if grains['reg_code'] %}
+register_system:
+  cmd.run:
+    - name: /usr/bin/SUSEConnect -r {{ grains['reg_code'] }}  {{ ("-e " + str(grains['reg_email'])) if grains['reg_email'] != "" else "" }}
```
Comment (arbulu89) [inline salt/default/registration.sls]: This operation is failling in my case. The "str" operator does not exist

My suggestion is to update this line to:
- name: /usr/bin/SUSEConnect -r {{ grains['reg_code'] }}  {{ ("-e " ~ grains['reg_email']) if grains['reg_email'] != "" else "" }}

Here more info: https://stackoverflow.com/questions/9856576/casting-ints-to-str-in-jinja2

File: salt/default/registration.sls [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+{% endif %}
+
+{% if grains['reg_additional_modules'] %}
+{% for module, mod_reg_code in grains['reg_additional_modules'].items() %}
+{{ module }}_registration:
+  cmd.run:
+    - name: /usr/bin/SUSEConnect -p {{ module }}  {{ ("-r " + str(mod_reg_code)) if mod_reg_code != "" else "" }}
```
Comment (arbulu89) [inline salt/default/registration.sls]: Same problem here

**Comment by diegoakechi** [comment]
Thanks @arbulu89. This looks a much better option.

File: libvirt/terraform/README.md [hunk 3]
```diff
@@ -69,6 +69,15 @@ additional_repos = {
 }
 "cluster_ssh_pub = salt://hana_node/files/sshkeys/my_key.id_rsa.pub"
 "cluster_ssh_key = salt://hana_node/files/sshkeys/my_key.id_rsa"
+
+# Optional SUSE Customer Center Registration parameters
+reg_code = "<<REG_CODE>>"
+reg_email = "<<your email>>"
+reg_additional_modules = {
+    "sle-module-adv-systems-management/12/x86_64" = ""
+   "sle-module-containers/12/x86_64" = ""
```
Comment (ldevulder) [inline libvirt/terraform/README.md]: Just a missing space here :)

**Comment by hsehic** [comment]
@diegoakechi @arbulu89 guys could you please look into this conflicts, thanks

**Comment by arbulu89** [comment]
@hsehic Conflicts resolved!

**Comment by arbulu89** [comment]
Upstream master merged to solver libvirt travis issues and files formatted to follow terraform standards

**Pull request merged**